receiver/mongodbatlasreceiver/receiver_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlasreceiver
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+)
+
+func TestDefaultConfig(t *testing.T) {
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig().(*Config)
+	params := componenttest.NewNopReceiverCreateSettings()
+	ctx := context.Background()
+
+	receiver, err := createMetricsReceiver(
+		ctx,
+		params,
+		cfg,
+		consumertest.NewNop(),
+	)
+	require.NoError(t, err)
+	require.NotNil(t, receiver, "receiver creation failed")
+
+	err = receiver.Start(ctx, componenttest.NewNopHost())
+	require.NoError(t, err)
+
+	err = receiver.Shutdown(ctx)
+	require.NoError(t, err)
+}
+
+func TestTimeConstraints(t *testing.T) {
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig().(*Config)
+
+	recv := receiver{
+		cfg: cfg,
+	}
+
+	tt := []struct {
+		name string
+		run  func(t *testing.T)
+	}{
+		{
+			name: "initial lookback is now() - collection_interval",
+			run: func(t *testing.T) {
+				now := time.Now()
+				tc := recv.timeConstraints(now)
+				require.NotNil(t, tc)
+				require.Equal(t, tc.start, now.Add(cfg.CollectionInterval*-1).UTC().Format(time.RFC3339))
+				// set lookback for next test after this one is done
+				recv.lastRun = now
```

Comment:
It doesn't seem right to make one test case dependent on another. I would recommend extracting subtests into a plain test case or making them independent.

Reply:
I was thinking the same thing when I wrote it tbh, I'll split it out now.

Reply:
@dmitryax updated to make them independent.